pkg/controller/deployment/sync.go

```diff
@@ -292,25 +372,22 @@ func (dc *DeploymentController) getNewReplicaSet(deployment *extensions.Deployme
 	deploymentutil.SetNewReplicaSetAnnotations(deployment, &newRS, newRevision, false)
 	createdRS, err := dc.client.Extensions().ReplicaSets(namespace).Create(&newRS)
 	if err != nil {
-		return nil, fmt.Errorf("error creating replica set %v: %v", deployment.Name, err)
+		// TODO: Make sure this condition will be aded only once! Subsequent tries to add the same condition
+		// should not overwrite.
+		condition := deploymentutil.NewCondition(extensions.DeploymentProgressing, api.ConditionUnknown, "NewReplicaSet", fmt.Sprintf("Failed to create new replica set %q: %v", createdRS.Name, err))
+		deployment.Status, _ = deploymentutil.SetCondition(deployment.Status, *condition, true)
+		dc.client.Extensions().Deployments(deployment.ObjectMeta.Namespace).UpdateStatus(deployment)
```

Comment:
We don't care about it enough but I can at least log it. The create error will be retried and I am going to add logic to retry adding the condition if this update fails.